htdocs/js/ui/discovery_page.js

```diff
@@ -23,59 +21,63 @@ RCloud.UI.discovery_page = (function() {
                       .first(20)
                       .map(function(notebook) {
                         var current = editor.get_notebook_info(notebook[0]);
-                        return {
-                          id: notebook[0],
-                          time: notebook[1],
-                          description: current.description,
-                          last_commit: new Date(current.last_commit).toDateString(),
-                          username: current.username,
-                          num_stars: editor.num_stars(current[0]),
-                          image_url: 'notebook.R/' + notebook[0] + '/thumb.png'
-                        }
+                        return rcloud.get_thumb(notebook[0]).then(function(thumb_src){
+                          return Promise.resolve({
+                            id: notebook[0],
+                            time: notebook[1],
+                            description: current.description,
+                            last_commit: new Date(current.last_commit).toDateString(),
+                            username: current.username,
+                            num_stars: editor.num_stars(current[0]),
+                            image_src: "data:image/png;base64," + thumb_src
+                          })
+                        });
```

Comment:
I don't think `Promise.resolve` is required here. A function returning inside a promise chain is always returning a resolved value.